src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -93,21 +96,24 @@ public void updateFilteredPatientList(Predicate<Patient> predicate) {
     }
 
     @Override
-    public Optional<Doctor> getDoctorIfPresent() {
-        Optional<Doctor> optionalDoctor = Optional.empty();
-        if (!model.getFilteredDoctorList().isEmpty()) {
-            optionalDoctor = Optional.of(model.getFilteredDoctorList().get(0));
-        }
-        return optionalDoctor;
+    public void updateFilteredPatientListBasedOnDoctor(Doctor doctor) {
+        requireNonNull(doctor);
+        PatientFilter patientContainsDoctor =
+                new PatientFilter(doctor);
+        Predicate<Patient> patientsOfDoctorPredicate =
+                new PatientContainsKeywordsPredicate(patientContainsDoctor);
+        this.updateFilteredPatientList(patientsOfDoctorPredicate);
```

Comment:
Is there any reason for replacing getDoctorIfPresent with updateFilteredPatientList?

Reply:
It's not a replacement; the `getDoctorIfPresent` function is not being used